Snapshot for `-profile test_single_end`

dialvarezs opened the pull request:
Closes #902.

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/main/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core pipelines lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

dialvarezs committed:
test_single_end snapshot [WIP]

dialvarezs committed:
Fix linting

dialvarezs committed:
Merge branch 'dev' into snapshot_single_end

dialvarezs committed:
Remove duplicated lines

dialvarezs committed:
Add genome binning section

dialvarezs committed:
Update snapshots

dialvarezs committed:
Add annotation dir

dialvarezs committed:
Add taxonomy dir

dialvarezs committed:
Add virus identification dir

dialvarezs committed:
Add ancient dna dir

dialvarezs committed:
Update changelog

dialvarezs committed:
Exclude more unstable files

dialvarezs committed:
Exclude all VCFs

dialvarezs committed:
Merge branch 'dev' into snapshot_single_end

dialvarezs committed:
Merge remote-tracking branch 'upstream/dev' into snapshot_single_end

dialvarezs committed:
Update snapshot

dialvarezs committed:
Move removal of quast_bins_summary to deprecation section in the changelog

dialvarezs committed:
Merge remote-tracking branch 'upstream/dev' into snapshot_single_end

dialvarezs committed:
Update snapshot

dialvarezs committed:
Merge remote-tracking branch 'upstream/dev' into snapshot_single_end

dialvarezs committed:
Fix linting

jfy133 commented on tests/.nftignore:
```diff
@@ -1,6 +1,12 @@
 .DS_Store
+Ancient_DNA/pydamage/analyze/*/*.csv
+Ancient_DNA/variant_calling/*/*.vcf.gz
+Annotation/Prokka/**/*.{log,err,gbk,sqn}
```
No addition checks for gbk/sqn? These are common downstream files, so might be good to validate somehow

jfy133 commented on tests/.nftignore:
```diff
@@ -1,6 +1,12 @@
+Annotation/Prokka/**/*.tmp.*
 Assembly/MEGAHIT/*.log
 Assembly/MEGAHIT/QC/*/*.bowtie2.log
+Assembly/MEGAHIT/QC/*/QUAST/*.{pdf,html,log}
```
To verify, there is no 'nice' 'completed successful' entry in the `log` file? Other two are fine

jfy133 commented on tests/.nftignore:
```diff
@@ -17,8 +23,13 @@ GenomeBinning/QC/BUSCO/**/*.log
+GenomeBinning/QC/CheckM/*/storage/*
 GenomeBinning/QC/CheckM2/**/DIAMOND_RESULTS.tsv
 GenomeBinning/QC/CheckM2/*/checkm2.log
+GenomeBinning/QC/QUAST/*/*.{pdf,html,log}
```
To verify, there is no 'nice' 'completed successful' entry in the log file? Other two are fine

jfy133 reviewed (commented):
Overall LGTM, 

As a general thing, I suggest wrapping _all_ asserts in `snapshot` even if it's just a boolean (you can also store these in a string even for more human readability, see nf-core/createtaxdb as an example -inspired by @maxulysse )

If that particular check fails and is not in a snapshot, it can be really annoying to debug.

I have 3 minor things that may have been missed, but otherwise this is looking very comprehensive, great work @dialvarezs !

dialvarezs committed:
Merge branch 'dev' into snapshot_single_end

dialvarezs committed:
Update snapshots

dialvarezs committed:
Merge remote-tracking branch 'upstream/dev' into snapshot_single_end

dialvarezs committed:
chore: update snapshots

dialvarezs committed:
feat: add boolean to snapshot, and check additional files

dialvarezs committed:
chore: fix changelog

dialvarezs committed:
fix: exclude all VCFs from snapshot

dialvarezs commented:
I came back to finish this finally! @jfy133

I addressed your comments:
- Booleans are included in snapshots with messages to indicate what are they testing, and also applied the same idea to `summary-rows` and `log-checks`
- VCFs are not stable because they include the date or datetime in the header, so no snapshots for them. But I added if they are not empty using `nft-vcf`.
- Added GBK and SQN checks. I intented to leave GBK and SQN in the snapshot, but they also have the date encoded, so they're not stable.
- `*_aggregated_classification.tsv` is not stable either, but we can keep the non-empty check.

dialvarezs committed:
chore: move workflow.sucess inside assertAll

dialvarezs committed:
fix: no comments in nftignore

dialvarezs committed:
chore: exclude sqn and gbk too

dialvarezs committed:
Merge remote-tracking branch 'upstream/dev' into snapshot_single_end

jfy133 commented on tests/test_single_end.nf.test:
```diff
@@ -10,36 +10,187 @@ nextflow_pipeline {
+                    "Prokka GBK files contain LOCUS: ${prokka_gbk.collect { file -> path("${file}").text.contains('LOCUS') }.every()}",
+                    "Prokka SQN files non-empty: ${prokka_sqn.collect { file -> path("${file}").size() > 0 }.every()}",
+                    // Summary row count checks
+                    "Bin summary non-empty: ${bin_summary.csv(sep: '\t').rowCount > 0}",
```
Could this not still be the case even if there are no hits, because you still have the header? Maybe minimum `>1`?

jfy133 replied:
Same for all CSV files

dialvarezs replied:
nft-csv plugin doesn't count the header.

jfy133 commented on CHANGELOG.md:
```diff
@@ -26,6 +28,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### `Deprecated`
 
+- [#908](https://github.com/nf-core/mag/pull/908) - Removed local `quast_bins_summary` in favor of `csvtk/concat` (by @dialvarezs)
```
Note If we follow semantic versioning this requires v6 btw, as a user can not longer use a config that points to thw QUAST_BIN_SUMMARY module

dialvarezs replied:
In theory yes, but I consider this one a soft one, because it's a replacement and normally users won't change the config for this (it's only a CSV concat after all).
Not like when we removed raw read taxonomic profiling, that one definitively deserved a major bump.

In dev there is also the removal of MAG_DEPTHS_PLOT, that's definitively a more serious breaking change.


But... i we adhere to strict semantic versioning, we'll end up with mag v10 on 2027 😂

jfy133 reviewed (approved):
One suggestion but not a blocker
